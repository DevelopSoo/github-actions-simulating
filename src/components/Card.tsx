export default function Card({
  title,
  description,
  imageUrl,
}: {
  title: string;
  description: string;
  imageUrl: string;
}) {
  return (
    <div className="rounded-lg bg-white p-4 shadow-md">
      <h2>{title}</h2>
      <p className="text-sm text-gray-600">{description}</p>
      <img src={imageUrl} alt={title} className="h-40 w-full object-cover" />
    </div>
  );
}
