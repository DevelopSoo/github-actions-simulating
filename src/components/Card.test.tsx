// src/components/Card.test.tsx

import { render } from "@testing-library/react";
import Card from "./Card";

test("Card 컴포넌트가 정보를 올바르게 표시합니다", () => {
  const cardProps = {
    title: "제목",
    description: "설명",
    imageUrl: "https://example.com/image.jpg",
  };

  const { container } = render(<Card {...cardProps} />);
  expect(container).toMatchSnapshot();
});
