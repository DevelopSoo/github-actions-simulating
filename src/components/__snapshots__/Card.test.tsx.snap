// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 컴포넌트가 정보를 올바르게 표시합니다 1`] = `
<div>
  <div
    class="rounded-lg bg-white p-4 shadow-md"
  >
    <h2>
      제목
    </h2>
    <p
      class="text-sm text-gray-600"
    >
      설명
    </p>
    <img
      alt="제목"
      class="h-40 w-full object-cover"
      src="https://example.com/image.jpg"
    />
  </div>
</div>
`;
